src/ImageSharp/MetaData/Profiles/ICC/IccProfile.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System;
+    using System.Collections.Generic;
+#if !NETSTANDARD1_1
+    using System.Security.Cryptography;
+#endif
+
+    /// <summary>
+    /// Represents an ICC profile
+    /// </summary>
+    public sealed class IccProfile
+    {
+        /// <summary>
+        /// The byte array to read the ICC profile from
+        /// </summary>
+        private byte[] data;
+
+        /// <summary>
+        /// The backing file for the <see cref="Entries"/> property
+        /// </summary>
+        private List<IccTagDataEntry> entries;
+
+        /// <summary>
+        /// ICC profile header
+        /// </summary>
+        private IccProfileHeader header;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IccProfile"/> class.
+        /// </summary>
+        public IccProfile()
+            : this((byte[])null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IccProfile"/> class.
+        /// </summary>
+        /// <param name="data">The raw ICC profile data</param>
+        public IccProfile(byte[] data)
+        {
+            this.data = data;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IccProfile"/> class
+        /// by making a copy from another ICC profile.
+        /// </summary>
+        /// <param name="other">The other ICC profile, where the clone should be made from.</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="other"/> is null.</exception>>
+        public IccProfile(IccProfile other)
+        {
+            Guard.NotNull(other, nameof(other));
+
+            // TODO: Do we need to copy anything else?
+            this.data = other.data;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IccProfile"/> class.
+        /// </summary>
+        /// <param name="header">The profile header</param>
+        /// <param name="entries">The actual profile data</param>
+        internal IccProfile(IccProfileHeader header, IEnumerable<IccTagDataEntry> entries)
+        {
+            Guard.NotNull(header, nameof(header));
+            Guard.NotNull(entries, nameof(entries));
+
+            this.header = header;
+            this.entries = new List<IccTagDataEntry>(entries);
+        }
+
+        /// <summary>
+        /// Gets or sets the profile header
+        /// </summary>
+        public IccProfileHeader Header
+        {
+            get
+            {
+                this.InitializeHeader();
+                return this.header;
+            }
+
+            set => this.header = value;
+        }
+
+        /// <summary>
+        /// Gets the actual profile data
+        /// </summary>
+        public List<IccTagDataEntry> Entries
```

Comment:
I'm totally unfamiliar with the ICC stuff, and I'm not sure if it's worth to overthink it's API, but considering the fact that the API is public I think it's worth to ask a few questions:

- Are we sure the lazy initialization way is the best way here? For a big header it could produce an overhead on an unexpected point from the users POV (reading a property).
- Is it really a good way to expose this stuff through `List<T>`? Do intend it to be writable?